### DISCUSSION TITLE
Lambda Preset Docs end abruptly

### DISCUSSION THREAD

**Discussion by siolfyr** [nuxt/nuxt#16909] [Questions]
Also, the docs link to Amazon Lex? That's not intentional, right?

https://v3.nuxtjs.org/docs/deployment/presets/lambda

What should the lambda function handler look like? The example imports handler from ./.output/server, but Lambda functions don't normally use imports...? Not sure what the intended usage is. If I try to make the Lambda function count as a "module" by adding `"type": "module"` to package.json I get `ERR_UNSUPPORTED_DIR_IMPORT` ...?

**Comment by joostdelange** [comment]
I've been struggeling with this for a while now too.
What I don't understand with the lambda preset is how to use it in an actual use case.

There are lots of examples on the internet like
- Cloudfront with Lambda@edge
- Lambda with an Api Gateway proxy

But I've had no success with the handler that is being generated using this preset.
I've tried doing the following, mainly because the entry file is an `index.mjs` which lambda does not support:
- Convert the `.mjs` file to a `.js` using esbuild so lambda can use it
- create my own handler which imports the `.mjs` asynchronous and then returns it like so:
```
module.exports.handler(async (event, context) => {
  const { handler } = await import('./.output/server/index.mjs')

  return handler
})
```
And this works partially but I don't think you should need to do this.
My guess is the `nitro` -> `lambda` preset should output the code in such a way so lambda functions can actually read it.

I'd really like to hear it if other people have been struggling with this and maybe found a solution.

**Comment by dan-danciu** [comment]
How is the lambda preset supposed to work since it exports ES6 syntax while it is clearly stated in the AWS documentation that:
Lambda does not currently support ECMAScript 6 (ES6) syntax.

**Comment by joostdelange** [comment]
Update: AWS Lambda now supports ES modules: [https://aws.amazon.com/blogs/compute/using-node-js-es-modules-and-top-level-await-in-aws-lambda](url)

This means the output which the Nitro engine creates (es6 syntax with a `.mjs` extension) is usuable inside AWS Lambda with Node.js 14 runtimes 🚀

**Reply by steffenstolze** [reply]
@joostdelange did you manage to deploy a Nuxt 3 App with server routes to AWS, utilizing API Gateway and Lambda?

**Reply by joostdelange** [reply]
@steffenstolze yes I did. Using the Nuxt 3 Nitro preset **lambda** you can build your application for AWS serverless. If you then connect that to a Lambda function using the _serverless-http_ package and express, you can connect that Lambda to an API Gateway API with its routes set to `/{proxy+}`

I'm happy to help you with other questions later, if you have any
